bluemira/equilibria/coils/_grouping.py

```diff
@@ -515,7 +515,9 @@ def _get_coiltype(self, ctype):
 
     def get_coiltype(self, ctype: Union[str, CoilType]):
         """Get coil by coil type"""
-        return CoilGroup(*self._get_coiltype(ctype))
+        if coiltype := self._get_coiltype(ctype):
+            return type(self(*coiltype))
```

Comment:
Bracket in the wrong place it looks like
```suggestion
            return type(self)(*coiltype)
```

Reply:
Thinking about this (I know I suggested this) I think it may be better to add an override to `CoilSet` that looks like this:

```python
    # Coilset
    def get_coiltype(self, ctype): 
        """Get coils by coils type"""
        if coiltype := self._get_coiltype(ctype):
            return CoilSet(*coiltype) 
        return None
```
and this change
```suggestion
            return CoilGroup(*coiltype)
```
I reckon they're could be some hidden bugs when `Circuit` is used otherwise.